src/util.cpp

```diff
@@ -394,6 +394,44 @@ void AdaptiveScheduler::reset() {
   start_time = std::chrono::steady_clock::now();
 }
 
+ProgressMonitor::ProgressMonitor(int64_t target_seconds,
+                                 const std::string &progress_file)
+    : start_time(std::chrono::steady_clock::now()),
+      target_seconds(target_seconds),
+      progress_file(progress_file) {
+  if (target_seconds <= 0) {
+    throw std::runtime_error("internal error: invalid target duration: " +
+                             std::to_string(target_seconds));
+  }
+}
```

Comment:
Reject target_seconds that are too far into the future, potentially causing undefined behavior or wrap around.

Reply:
difficult to define a meaningful upper limit